src/VisualStudio/Core/Def/ServicesVSResources.resx

```diff
@@ -1332,4 +1332,13 @@ I agree to all of the foregoing:</value>
   <data name="A_new_editorconfig_file_was_detected_at_the_root_of_your_solution_Would_you_like_to_make_it_a_solution_item" xml:space="preserve">
     <value>A new .editorconfig file was detected at the root of your solution. Would you like to make it a solution item?</value>
   </data>
+  <data name="Run_Code_Analysis_on_0" xml:space="preserve">
+    <value>Run Code Analysis on {0}</value>
```

Comment:
This needs `'` to avoid the situation in #39411. As an alternative, separate resource strings can be used for solution and project scenarios.

Reply:
This matches the exact syntax of all existing "Run Code Analysis on ProjectName" commands at different places. I would like us to stay consistent here.

Reply:
We should separate "Run Code Analysis on Solution" to a separate string then

Reply:
Note that this is for an error path/defensive code. I am not sure if we can reach a situation where both project.FilePath and solution.FileName are null. I would still go the route you suggested.